Minesweeper.java

```diff
@@ -142,23 +200,23 @@ public Game(int gridSize, int bombAmount){
 
             // must be done after bomb population 
             // as bombs have a separate action listener
-            for (int y = 0; y < this.gridSize; y++) {
-                for (int x = 0; x < this.gridSize; x++) {
+            for (int y = 0; y < Game.gridSize; y++) {
+                for (int x = 0; x < Game.gridSize; x++) {
                     final Cell cell = cells[y][x];
-                    final Game self = this;
 
                     if (!cell.isBomb) {
+                        final int x2 = x;
```

Comment:
there has to be a better way of doing this right? this is not only counterintuitive but also hard to read: "why is a variable named `x2` equal to `x`?".

Reply:
since the actionlistener is an inner "class" when variables outside it, i.e the counters from the loops y and x are given to it as parameters in the method, it wants them to be final or effectively final, thats why those two new variables are created, i agree that we can/should rename them

Reply:
is it possible to do `final int x = x`?

Reply:
im not really sure, i dont know if java would like the same variable name and if it will distinguish between the two. we could just call them xFinal/yFinal

Reply:
addressed in latest commit